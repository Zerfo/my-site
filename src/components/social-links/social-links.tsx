import cn from 'classnames';

import styles from './styles.module.scss';

export const SocialLinks = () => {
  return (
    <div className={styles.socialLinks} data-testid="social-links">
      <a href="https://moikrug.ru/nikolaysmekalin" target="_blank">
        <div
          className={cn(styles.socialLinks_icon, styles.socialLinks_cv)}
          data-testid="social-link"
        />
      </a>
      <a href="https://github.com/Zerfo" target="_blank">
        <div
          className={cn(styles.socialLinks_icon, styles.socialLinks_git)}
          data-testid="social-link"
        />
      </a>
      <a href="https://t.me/zerfo" target="_blank">
        <div
          className={cn(styles.socialLinks_icon, styles.socialLinks_tg)}
          data-testid="social-link"
        />
      </a>
      <a href="https://vk.com/nik.smekalin" target="_blank">
        <div
          className={cn(styles.socialLinks_icon, styles.socialLinks_vk)}
          data-testid="social-link"
        />
      </a>
      <a href="https://www.instagram.com/nik.smekalin" target="_blank">
        <div
          className={cn(styles.socialLinks_icon, styles.socialLinks_inst)}
          data-testid="social-link"
        />
      </a>
    </div>
  );
};
