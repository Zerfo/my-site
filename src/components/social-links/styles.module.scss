.socialLinks {
  display: flex;
  width: 256px;
  justify-content: space-between;
  align-items: center;

  &_icon {
    width: 32px;
    height: 32px;
    background-size: 32px 32px;
    cursor: pointer;
    transition-timing-function: ease-out;
    transition: 0.2s;

    &:hover {
      transform: scale(1.1);
      transition-timing-function: ease-in;
      transition: 0.2s;
    }
  }

  &_cv {
    background-image: url('../../assets/icons/resume.png');
  }
  &_git {
    background-image: url('../../assets/icons/github.png');
  }
  &_tg {
    background-image: url('../../assets/icons/telegram.png');
  }
  &_vk {
    background-image: url('../../assets/icons/vk.png');
  }
  &_inst {
    background-image: url('../../assets/icons/instagram.png');
  }
}