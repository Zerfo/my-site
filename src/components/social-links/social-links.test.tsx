import { render, screen } from '@testing-library/react';

import { SocialLinks } from './social-links';

describe('components/Footer', () => {
  it('Компонент должен соответствовать снэпшоту', () => {
    const socialLinksEl = render(<SocialLinks />);

    expect(socialLinksEl).toMatchSnapshot();
  });

  it('У компонента должно быть 6 иконок', async () => {
    render(<SocialLinks />);

    const socialLinksElements = await screen.findAllByTestId('social-link');

    expect(socialLinksElements).toHaveLength(5);
  });
});
