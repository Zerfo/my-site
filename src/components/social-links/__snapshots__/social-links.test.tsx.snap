// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Footer Компонент должен соответствовать снэпшоту 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="socialLinks"
        data-testid="social-links"
      >
        <a
          href="https://moikrug.ru/nikolaysmekalin"
          target="_blank"
        >
          <div
            class="socialLinks_icon socialLinks_cv"
            data-testid="social-link"
          />
        </a>
        <a
          href="https://github.com/Zerfo"
          target="_blank"
        >
          <div
            class="socialLinks_icon socialLinks_git"
            data-testid="social-link"
          />
        </a>
        <a
          href="https://t.me/zerfo"
          target="_blank"
        >
          <div
            class="socialLinks_icon socialLinks_tg"
            data-testid="social-link"
          />
        </a>
        <a
          href="https://vk.com/nik.smekalin"
          target="_blank"
        >
          <div
            class="socialLinks_icon socialLinks_vk"
            data-testid="social-link"
          />
        </a>
        <a
          href="https://www.instagram.com/nik.smekalin"
          target="_blank"
        >
          <div
            class="socialLinks_icon socialLinks_inst"
            data-testid="social-link"
          />
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="socialLinks"
      data-testid="social-links"
    >
      <a
        href="https://moikrug.ru/nikolaysmekalin"
        target="_blank"
      >
        <div
          class="socialLinks_icon socialLinks_cv"
          data-testid="social-link"
        />
      </a>
      <a
        href="https://github.com/Zerfo"
        target="_blank"
      >
        <div
          class="socialLinks_icon socialLinks_git"
          data-testid="social-link"
        />
      </a>
      <a
        href="https://t.me/zerfo"
        target="_blank"
      >
        <div
          class="socialLinks_icon socialLinks_tg"
          data-testid="social-link"
        />
      </a>
      <a
        href="https://vk.com/nik.smekalin"
        target="_blank"
      >
        <div
          class="socialLinks_icon socialLinks_vk"
          data-testid="social-link"
        />
      </a>
      <a
        href="https://www.instagram.com/nik.smekalin"
        target="_blank"
      >
        <div
          class="socialLinks_icon socialLinks_inst"
          data-testid="social-link"
        />
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
